src/lib/pmd/OutputProcessor.ts

```diff
@@ -32,42 +33,46 @@ export class OutputProcessor extends AsyncCreatable {
     // We want to find any events that were dumped into stdout or stderr and turn them back into events that can be thrown.
     // As per the convention outlined in SfdxMessager.java, SFDX-relevant messages will be stored in the outputs as JSONs
     // sandwiched between 'SFDX-START' and 'SFDX-END'. So we'll find all instances of that.
-    public processOutput(stdout: string, stderr: string): void {
-        this.logger.trace(`stdout: ${stdout}`);
-        this.logger.trace(`stderr: ${stderr}`);
+    public processOutput(out: string): void {
+        this.logger.trace(`stdout: ${out}`);
+        if(!out) {
+            // Nothing to do here
+            return;
+        }
 
-        const outEvents: PmdCatalogEvent[] = this.getEventsFromString(stdout);
-        // TODO: After code streamlining, Main.java prints all messages at once to stdout. Consider removing stderr processing
-        const errEvents: PmdCatalogEvent[] = this.getEventsFromString(stderr);
-        this.logger.trace(`Total count of events found: stdout - ${outEvents.length}, stderr - ${errEvents.length}`);
+        const outEvents: PmdCatalogEvent[] = this.getEventsFromString(out);
+        this.logger.trace(`Total count of events found: ${outEvents.length}`);
 
-        this.orderAndEmitEvents(outEvents, errEvents);
+        this.emitEvents(outEvents);
     }
 
     // TODO: consider moving all message creation logic to a separate place and making this method private
-    public orderAndEmitEvents(outEvents: PmdCatalogEvent[], errEvents: PmdCatalogEvent[]): void {
+    public emitEvents(outEvents: PmdCatalogEvent[]): void {
         this.logger.trace('About to order and emit');
-        // If both lists are empty, we can just be done now.
-        if (outEvents.length == 0 && errEvents.length == 0) {
+        // If list is empty, we can just be done now.
+        if (outEvents.length == 0) {
             return;
         }
 
-        // If either list is non-empty, we'll need to interleave the events in both lists into a single list that's ordered
-        // chronologically. We'll do that with a bog-standard merge algorithm.
-        const orderedEvents = this.orderEventsChronologically(outEvents, errEvents);
-
         // Iterate over all of the events and throw them as appropriate.
-        orderedEvents.forEach((event) => {
+        outEvents.forEach((event) => {
+            this.logEvent(event);
             if (event.handler === 'UX') {
                 const eventType = `${event.type.toLowerCase()}-${event.verbose ? 'verbose' : 'always'}`;
-                this.logger.trace(`Sending new event of type ${eventType} and message ${event.key}`);
-                this.logEvent(event);
-                uxEvents.emit(eventType, messages.getMessage(event.key, event.args));
+                this.emitUxEvent(eventType, event.messageKey, event.args);
+            } else if (event.handler === 'INTERNAL' && event.type === 'ERROR') {
```

Comment:
Do we need a final `else` here?

Reply:
Line 69 would log event for all incoming events. If we get an event with `INTERNAL` that's not an `ERROR`, I feel logging the event should be sufficient. Let me know if you have other ideas.